app/components/post_textbox/post_textbox_base.js

```diff
@@ -408,62 +414,154 @@ export default class PostTextBoxBase extends PureComponent {
-                // message wraps, for example), then when the text is cleared
-                // the PostTextbox height decrease will be animated. This
-                // animation in conjunction with the PostList animation as it
-                // receives the newly created post is causing issues in the iOS
-                // PostList component as it fails to properly react to its content
-                // size changes. While a proper fix is determined for the PostList
-                // component, a small delay in triggering the height decrease
-                // animation gives the PostList enough time to first handle content
-                // size changes from the new post.
-                setTimeout(() => {
-                    this.handleTextChange('');
-                }, 250);
+                this.showSendToAllOrChannelAlert(currentMembersCount);
             } else {
-                this.handleTextChange('');
+                this.doSubmitMessage();
             }
+        }
+    };
 
-            this.changeDraft('');
+    textContainsAtAllAtChannel = (text) => {
+        const textWithoutCode = text.replace(/(`+)([^`]|[^`][\s\S]*?[^`])\1(?!`)| *(`{3,}|~{3,})[ .]*(\S+)? *\n([\s\S]*?\s*)\3 *(?:\n+|$)/g, '');
+        return (/\B@(all|channel)\b/i).test(textWithoutCode);
+    }
 
-            let callback;
-            if (Platform.OS === 'android') {
-                // Fixes the issue where Android predictive text would prepend suggestions to the post draft when messages
-                // are typed successively without blurring the input
-                const nextState = {
-                    keyboardType: 'email-address',
-                };
+    showSendToAllOrChannelAlert = (currentMembersCount) => {
+        const {intl} = this.context;
+        const {channelTimezoneCount} = this.state;
+        const {isTimezoneEnabled} = this.props;
+        const notifyAllTitle = (
+            intl.formatMessage({
+                id: 'mobile.post_textbox.entire_channel.title',
+                defaultMessage: 'Confirm sending notifications to entire channel',
+            })
+        );
 
-                callback = () => this.setState({keyboardType: 'default'});
+        const notifyAllConfirm = (
+            intl.formatMessage({
+                id: 'mobile.post_textbox.entire_channel.confirm',
+                defaultMessage: 'Confirm',
+            })
+        );
 
-                this.setState(nextState, callback);
-            }
+        const notifyAllCancel = (
+            intl.formatMessage({
+                id: 'mobile.post_textbox.entire_channel.cancel',
+                defaultMessage: 'Cancel',
+            })
+        );
+        let notifyAllMessage = '';
+        if (isTimezoneEnabled && channelTimezoneCount) {
+            notifyAllMessage = (
+                intl.formatMessage(
+                    {
+                        id: 'mobile.post_textbox.entire_channel.message',
```

Comment:
This should have its own translation ID.

Reply:
Resolved! Thanks for pointing that out. Didn't catch it